### PR TITLE
feat!: Switch to new pedersen implementation

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

## Problem\*

Resolves 

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by kevaundray** [comment]
Note that I have not updated the integration tests, yet they are passing;

This is because those tests used a fixed version of barretenberg and are not using the pedersen/schnorr functions.

File: acvm-repo/barretenberg_blackbox_solver/src/wasm/pedersen.rs [hunk 1]
```diff
@@ -38,13 +38,14 @@ impl Pedersen for Barretenberg {
 #[test]
 fn pedersen_hash_to_point() -> Result<(), Error> {
     let barretenberg = Barretenberg::new();
-    let (x, y) = barretenberg.encrypt(vec![FieldElement::zero(), FieldElement::one()], 0)?;
+    let (x, y) = barretenberg
+        .encrypt(vec![FieldElement::from(1 as u128), FieldElement::from(1 as u128)], 1)?;
```
Comment (TomAFrench) [inline acvm-repo/barretenberg_blackbox_solver/src/wasm/pedersen.rs:42]: generator index 0 is forbidden now? Why did with switch to 1?

Comment (TomAFrench) [reply acvm-repo/barretenberg_blackbox_solver/src/wasm/pedersen.rs:42]: up to you :shrug:

**Review by sirasistant — APPROVED** [review]
LGTM

**Pull request merged**